cmd/support_archive/resources.go

```diff
@@ -7,13 +7,21 @@ import (
 	"strings"
 
 	"github.com/go-logr/logr"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/yaml"
 )
 
-const k8sResourceCollectorName = "k8sResourceCollector"
+const (
+	k8sResourceCollectorName = "k8sResourceCollector"
+	webhookValidatorName     = "dynatrace-webhook"
+	dynakubeCRDName          = "dynakubes.dynatrace.com"
+	edgeconnectCRDName       = "edgeconnects.dynatrace.com"
```

Comment:
just random thoughts: is it possible to collect all CRDS by api-group?
```
kubectl api-resources --api-group=dynatrace.com
NAME           SHORTNAMES   APIVERSION               NAMESPACED   KIND
dynakubes                   dynatrace.com/v1beta1    true         DynaKube
edgeconnects                dynatrace.com/v1alpha1   true         EdgeConnect
```

Reply:
why is it important, as soon as we split dynakube resource, we need to add each new resource here, instead of get all our CRDS.

cc @0sewa0 wdyt?

Reply:
thanks for the input, i will need to pass clientset to the k8scollector struct in order to do this. I will try because I think this is a nice solution.

Reply:
I did it now that way that I list the CRDs by api-group. Everything works, expect for the unit tests. There we have the problem that we probably need to mock this call to the api-group, which I am currently trying to do. Do you guys have a better solution there?

For example: https://github.com/Dynatrace/dynatrace-operator/blob/main/cmd/support_archive/resources_test.go#L134

Here I am adding the kube config, and this results then in the test failing because of refused connection to localhost:80. Should I try to fix these tests or should I stick to my other approach where I queried every CRD with .get() and not via api-group? 

The benefit would be that we would not need to rework the unit tests to such an extent, the downside would be that we would add a new api call for every new CRD that we add. 

What do you guys think?

Reply:
if you show us the problem we can help you to mock it :)

Reply:
pushed the changes in https://github.com/Dynatrace/dynatrace-operator/pull/2481/commits/c0fe2a95ffbb8275c0ed9a28818a7525c3e59ef9

Reply:
fixed in https://github.com/Dynatrace/dynatrace-operator/pull/2481/commits/7424836288a062b611295c5c7781b13cce58c5f6